spec/support/jazz.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# Here's the "application"
+module Jazz
+  # Here are some classes that the application deals with
+  module Models
+    Ensemble = Struct.new(:name)
+    Instrument = Struct.new(:name)
+  end
+
+  # Here's a new-style GraphQL type definition
+  class Ensemble < GraphQL::Object
+    model Models::Ensemble
+    description "A group of musicians playing together"
+    field :name, "String", null: false
+  end
+
+  # Lives side-by-side with an old-style definition
+  InstrumentType = GraphQL::ObjectType.define do
+    name "Instrument"
+    field :name, !types.String
+  end
+
+  # Another new-style definition, with method overrides
+  class Query < GraphQL::Object
+    field :ensembles, "[Ensemble]"
+    field :instruments, "[Instrument]"
+
+    def ensembles
```

Comment:
In our version we have `field` take an optional `resolve: ->(obj, args, ctx) { }`. The parameters are an important missing detail (does the caller check the arity here or something?) but otherwise I think I like this approach too. I think I'd suggest defining each method with its `field` call though, rather than putting all the `fields` at the top.